deepchecks/vision/suite.py

```diff
@@ -45,31 +40,24 @@ def run(
         self,
         train_dataset: Optional[VisionData] = None,
         test_dataset: Optional[VisionData] = None,
-        model: Optional[nn.Module] = None,
-        scorers: Optional[Mapping[str, Metric]] = None,
-        scorers_per_class: Optional[Mapping[str, Metric]] = None,
-        device: Union[str, torch.device, None] = None,
         random_state: int = 42,
         with_display: bool = True,
-        n_samples: Optional[int] = None,
-        train_predictions: Optional[Dict[int, Union[Sequence[torch.Tensor], torch.Tensor]]] = None,
-        test_predictions: Optional[Dict[int, Union[Sequence[torch.Tensor], torch.Tensor]]] = None,
-        train_properties: Optional[STATIC_PROPERTIES_FORMAT] = None,
-        test_properties: Optional[STATIC_PROPERTIES_FORMAT] = None,
-        model_name: str = '',
+        max_samples: Optional[int] = None,  # TODO: do something with this!
         run_single_dataset: Optional[str] = None
     ) -> SuiteResult:
         """Run all checks.
 
         Parameters
         ----------
-        train_dataset: Optional[VisionData] , default None
-            object, representing data an estimator was fitted on
-        test_dataset : Optional[VisionData] , default None
-            object, representing data an estimator predicts on
-        model : nn.Module , default None
-            A scikit-learn-compatible fitted estimator instance
-        {additional_context_params:2*indent}
+        train_dataset : Optional[VisionData] , default: None
+            VisionData object, representing data the model was fitted on
+        test_dataset : Optional[VisionData] , default: None
+            VisionData object, representing data the models predicts on
+        {additional_run_params:2*indent}
+        max_samples : Optional[int] , default: None
+            Maximum number of samples to use in checks. If none, each check will use either the full dataset or a
```

Comment:
It's always according to check n_samples. Only thing is that this can be None, in which case it'll run the full dataset.

Reply:
why? I do not agree